### ISSUE TITLE
Visual Eval Pipeline: Tier 2 Real-World Datasets

### ISSUE THREAD

**Issue by Trecek** [TalonT-Org/spectral-init#119]
## Summary

Add real-world dataset support (MNIST, Fashion-MNIST, Pendigits) to the visual evaluation pipeline, plus a placeholder hook for future single-cell data. These larger, real-world datasets provide a more rigorous stress test of the spectral initialization than synthetic data alone.

## Wave

**Wave 3** — depends on:
- #116 (Phase 1 Baseline Script & Scaffolding)

Tier 2 datasets use the same pipeline infrastructure from Phase 1. They will also work with Phase 2 comparison (#118) once that's merged.

### 1. Tier 2 Dataset Loaders

Extend the `load_dataset()` function in `tests/visual_eval/generate_umap_comparisons.py` to support:

| Dataset Name | Source | n | features | Classes | Notes |
|---|---|---|---|---|---|
| `mnist_10k` | `sklearn.datasets.fetch_openml('mnist_784')` | 10000 | 784 | 10 digits | Subsample first 10k; standard UMAP showcase |
| `fashion_mnist_10k` | `sklearn.datasets.fetch_openml('Fashion-MNIST')` | 10000 | 784 | 10 categories | Harder than MNIST; overlapping classes (shirt/coat/pullover) |
| `pendigits` | `sklearn.datasets.load_digits()` | 1797 | 64 | 10 digits | Small and fast; sklearn built-in (8x8 pixel images) |

**Note on pendigits**: The document mentions `pendigits (n=10992, 16 features)` from UCI, but `sklearn.datasets.load_digits()` provides a similar small-digit dataset with 1797 samples and 64 features. Use `load_digits()` as it requires no download. If the full UCI pendigits is desired, use `fetch_openml('pendigits')` instead — but `load_digits` is more practical and requires no network access.

#### MNIST/Fashion-MNIST Loading

```python
from sklearn.datasets import fetch_openml

def load_mnist(n_samples=10000):
    mnist = fetch_openml('mnist_784', version=1, as_frame=False)
    X, y = mnist.data[:n_samples].astype(np.float32), mnist.target[:n_samples].astype(int)
    return X, y, "mnist_10k"

def load_fashion_mnist(n_samples=10000):
    fmnist = fetch_openml('Fashion-MNIST', version=1, as_frame=False)
    X, y = fmnist.data[:n_samples].astype(np.float32), fmnist.target[:n_samples].astype(int)
    return X, y, "fashion_mnist_10k"
```

#### Dataset Registry

Use a dict/registry pattern so datasets are easy to add:

```python
DATASETS = {
    # Tier 1 (existing)
    "blobs_1000": load_blobs_1000,
    "circles_1000": load_circles_1000,
    # ...
    # Tier 2 (new)
    "mnist_10k": load_mnist,
    "fashion_mnist_10k": load_fashion_mnist,
    "pendigits": load_pendigits,
}
```

### 2. Tier 3 Single-Cell Placeholder

Add a placeholder interface for future single-cell data:

```python
def load_singlecell(path=None):
    """Load single-cell data from an AnnData .h5ad file.

    Requires scanpy: pip install scanpy
    Not implemented yet — placeholder for future use.

    Usage:
        X, labels, name = load_dataset("singlecell", path="data/pbmc3k.h5ad")
    """
    raise NotImplementedError(
        "Single-cell loading requires scanpy and a .h5ad file. "
        "Install scanpy and provide --singlecell-path to use this dataset."
    )
```

Add `--singlecell-path` CLI argument that's accepted but triggers the not-implemented error if no scanpy is installed.

### 3. Dataset Tier Selection

Add `--tier` CLI argument:

```bash
# Run only Tier 1 (fast, synthetic)
python generate_umap_comparisons.py --phase baseline --tier 1

# Run only Tier 2 (slower, real-world, requires download)
python generate_umap_comparisons.py --phase baseline --tier 2

# Run all (default)
python generate_umap_comparisons.py --phase baseline --tier all
```

### 4. Plot Adjustments for Large Datasets

Tier 2 datasets are larger (10k points) and higher-dimensional, so adjust plot parameters:
- Point size: 0.5-1 for n > 5000
- Alpha: 0.3 for n > 5000
- Consider adding dataset size to plot title: `"MNIST (n=10000) — Python Spectral Init"`

### 5. Caching for Downloaded Datasets

`fetch_openml` caches data in `~/scikit_learn_data/` by default. Document this in the README. Add a `--data-dir` argument to customize the cache location if desired, or just rely on sklearn's default caching.

## Implementation Notes

- `fetch_openml` may take time on first download — print a message like "Downloading MNIST (first time only)..."
- MNIST and Fashion-MNIST have 70k samples total; we subsample to 10k for practical runtime. Use deterministic subsampling: `rng = np.random.RandomState(42); idx = rng.choice(len(X), 10000, replace=False)`
- These datasets will be significantly slower to UMAP-fit than Tier 1 (minutes vs seconds). Print timing information.
- Fashion-MNIST is a good stress test because some classes genuinely overlap (shirt vs coat vs pullover) — spectral init should still outperform random
- For pendigits with `load_digits()`: data is already scaled 0-16, labels are 0-9. No preprocessing needed.

## Acceptance Criteria

- [ ] `--dataset mnist_10k` generates baseline plot and exports for MNIST
- [ ] `--dataset fashion_mnist_10k` generates baseline plot and exports for Fashion-MNIST
- [ ] `--dataset pendigits` generates baseline plot and exports for Pendigits
- [ ] `--tier 1`, `--tier 2`, `--tier all` work correctly
- [ ] `--dataset singlecell` raises a clear NotImplementedError with instructions
- [ ] Plots look good at n=10000 (small points, appropriate alpha)
- [ ] Dataset download/caching works correctly (sklearn default cache)
- [ ] Large datasets don't cause memory issues
- [ ] Timing info printed for each dataset

## Files Modified

- **Modify:** `tests/visual_eval/generate_umap_comparisons.py` (add Tier 2 loaders, tier selection, plot adjustments)
- **Modify:** `tests/visual_eval/README.md` (document Tier 2 datasets, download requirements)
